akka-stream-tests/src/test/scala/akka/stream/impl/TimeoutsSpec.scala

```diff
@@ -130,6 +130,105 @@ class TimeoutsSpec extends AkkaSpec {
 
   }
 
+  "BackpressureTimeout" must {
+
+    "pass through elements unmodified" in assertAllStagesStopped {
+      Await.result(Source(1 to 100).backpressureTimeout(500.millis).grouped(200).runWith(Sink.head), 3.seconds) should === (1 to 100)
+    }
+
+    "succeed if backpressure notified" in assertAllStagesStopped {
+      val subscriber = TestSubscriber.probe[Int]()
+
+      Source(1 to 4)
+        .backpressureTimeout(500.millis)
+        .runWith(Sink.fromSubscriber(subscriber))
+
+      for (i ← 1 to 3) {
+        subscriber.requestNext(i)
+        subscriber.expectNoMsg(100.millis)
+      }
+
+      subscriber.requestNext(4)
+      subscriber.expectComplete()
+      subscriber.expectNoMsg(1.second)
+    }
+
+    "not throw if publisher is less frequent then timeout" in assertAllStagesStopped {
+      val publisher = TestPublisher.probe[String]()
+      val subscriber = TestSubscriber.probe[String]()
+
+      Source.fromPublisher(publisher)
+        .backpressureTimeout(500.millis)
+        .runWith(Sink.fromSubscriber(subscriber))
+
+      subscriber.request(2)
+
+      subscriber.expectNoMsg(100.millis)
+      publisher.sendNext("Quick Msg")
+      subscriber.expectNext("Quick Msg")
+
+      subscriber.expectNoMsg(2.seconds)
```

Comment:
Yeah, in this test case I'd rather not, because I want to be sure about lack of failure.